python/triton/language/extra/cuda/_experimental_tma.py

```diff
@@ -0,0 +1,116 @@
+from triton.language import core
+from triton.language import semantic
+from triton._C.libtriton import ir
+
+from typing import Sequence
+
+__all__ = [
+    "experimental_device_tensormap_create1d",
+    "experimental_device_tensormap_create2d",
+    "experimental_tensormap_fenceproxy_acquire",
+]
+
+
+def _set_tma_dtype(desc_ptr: core.tensor, element_ty: core.dtype, _builder: ir.builder):
+    if element_ty.primitive_bitwidth == 8:
+        semantic.tensormap_replace_elemtype(desc_ptr, 0, _builder)
+    elif element_ty.primitive_bitwidth == 16:
+        semantic.tensormap_replace_elemtype(desc_ptr, 1, _builder)
+    elif element_ty.primitive_bitwidth == 32:
+        semantic.tensormap_replace_elemtype(desc_ptr, 2, _builder)
+    else:
+        raise ValueError("element_ty must be a primitive of size 1, 2, or 4 bytes but got")
+
+
+@core.builtin
+def experimental_device_tensormap_create1d(
+    desc_out: core.tensor,
+    template_desc: core.tensor,
+    global_address: core.tensor,
+    load_size: core.tensor,
+    global_size: core.tensor,
+    element_ty: core.dtype,
+    _builder: ir.builder,
+):
+    load_size = core._constexpr_to_value(load_size)
+    global_size = core._to_tensor(global_size, _builder)
+    element_ty = core._constexpr_to_value(element_ty)
+
+    local_desc = semantic.tensormap_allocate(template_desc, _builder)
```

Comment:
I know I had mentioned we should use templates before but since we are updating most of the fields anyway I wonder if we should just create the descriptor from scratch as it will most likely be more efficient. (See the discussion we had with Elliot on slack)